python/lsst/ts/standardscripts/maintel/m1m3/check_hardpoint.py

```diff
@@ -0,0 +1,89 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["CheckHardpoint"]
+
+from lsst.ts.idl.enums.MTM1M3 import HardpointTest
+from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
+
+from lsst.ts import salobj
+
+
+class CheckHardpoint(salobj.BaseScript):
+    """Check M1M3 Individual hardpoint breakaway.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    - "Testing hardpoint {id}.": Check hardpoint.
+    - "Hardpoint breakaway check complete.": Check complete.
+
+    """
+
+    def __init__(self, index, add_remotes: bool = True):
+        super().__init__(index=index, descr="Check M1M3 Hardpoint")
+
+        self.mtcs = MTCS(
+            domain=self.domain,
+            intended_usage=None if add_remotes else MTCSUsages.DryTest,
+            log=self.log,
+        )
+
+    @classmethod
+    def get_schema(cls):
+        return None
+
+    async def configure(self, config):
+        # This script does not require any configuration.
+        pass
+
+    def set_metadata(self, metadata):
+        metadata.duration = 180.0
+
+    async def wait_hardpoint_test_ok(self):
+        """Wait for the laser to warm up and be ready to be used."""
+
+        self.log.info("Checking if the bump test has passed.")
+
+        self.mtcs.rem.mtm1m3.evt_hardpointTestStatus.flush()
+        test_status = await self.mtcs.rem.mtm1m3.evt_hardpointTestStatus.aget(
+            timeout=self.timeout_std,
+        )
+
+        while test_status.status != HardpointTest.PASSED:
+            test_status = await self.mtcs.rem.mtm1m3.evt_hardpointTestStatus.next(
```

Comment:
This will fire asyncio.TimeoutError if message isn't received. You also enter endless loop in HardpointTest.FAILED, or if the CSC will keep sending messages without PASSED or FAILED. This need significant rework. Please see how timeouts are handled in ts_m1m3supporttesting/MTM1M3Movements and ts_m1m3supporttesting/M13T002.py (the T002, FA bump test, is much smarter about checking state change for hardpointTestStatus - CSC sometimes sends old message with PASSED/FAILED, so your loop for checking will not run as intended)

Reply:
Done, please see how it is done now to see if that seems okay. I used what was done in MTM1M3Movements